### DISCUSSION TITLE
Workspaces and Tabs

### DISCUSSION THREAD

**Discussion by juridiener** [wezterm/wezterm#3630] [Q&A]
Hello,
I have following issue with the first code example.
It doesn't create me the another workspace API.

And it doesn't select the left pane in the first tab of the workspace REACT.

```
wezterm.on("gui-startup", function()
local home = os.getenv("HOME")
local tab, pane, window = mux.spawn_window({
cwd = home .. "/Documents/react/",
workspace = "react",
})
window:gui_window():maximize()

tab:set_title("REACT")

-- Create a new tab in the react workspace
local ws_react_tab2 = window:spawn_tab({})
ws_react_tab2:set_title("tab2")

tab:activate()
pane:split({ direction = "Right", size = 0.1 })
pane.activate()

local ws_api_tab, ws_api_pane, ws_api_window = mux.spawn_window({
cwd = home .. "/Documents/api/",
workspace = "API",
})
ws_api_tab:set_title("API")
ws_api_pane.split({ direction = "Right", size = 0.1 })
ws_api_tab:activate()
ws_api_pane.activate()

-- We want to startup in the coding workspace
mux.set_active_workspace("react")
end)
```
But with this second code it creates the workspaces but still have that issue that the left pane is not selected.

```
wezterm.on("gui-startup", function()
local home = os.getenv("HOME")
local tab, pane, window = mux.spawn_window({
cwd = home .. "/Documents/react/",
workspace = "react",
})

window:gui_window():maximize()
tab:set_title("REACT")

-- Create a new tab in the dev workspace
local ws_react_tab2 = window:spawn_tab({})
ws_react_tab2:set_title("tab2")

local ws_api_tab, ws_api_pane, ws_api_window = mux.spawn_window({
cwd = home .. "/Documents/api/",
workspace = "API",
})
ws_api_tab:set_title("API")

tab:activate()
pane:split({ direction = "Right", size = 0.1 })
pane.activate()

local ws_api_tab2 = ws_api_window.spawn_tab({})
ws_api_tab2:set_title("Tab2")

ws_api_pane.split({ direction = "Right", size = 0.1 })
ws_api_tab:activate()
ws_api_pane.activate()

-- We want to startup in the coding workspace
mux.set_active_workspace("react")
end)
```

In gegeral what is the right order to create workspaces, 
then tabs and splits (panes) and then set them active.

Apparently the order is important.

And if I close the Terminal, i have to close it for all the workspaces, so in that case two times.
Is there a workaround that if I close the terminal it close all workspaces at the same time?
Sorry for that long code example.

**Comment by wez** [comment]
Just quickly glancing at this; I think you want `pane.activate()` to be `pane:activate()`

**Reply by juridiener** [reply]
Hey wez, thank you this was the problem. Sorry for that. I should have looked more closely.
Can you maybe Explain me what is the right order to create workspaces, tabs and panes and set them active? 
I tried this pattern, create a workspace, create tabs, create panes set tab active and panes. But that didn't work out.
Only if i create first all worspaces and then. Why is that so? 

Why i have to close first all workspace to close wezterm. Is there a way if to avoid this and close everything if i click on the X-Button?

Thanks in advance. AND BTW: wezterm is awesome? I like that i configure it how i want it and that the search is one of the best from all terminals i test it out.
